### ISSUE TITLE
about fine tuning kinetics 600 dataset

### ISSUE THREAD

**Issue by chyueG** [facebookresearch/SlowFast#49]
hi,I use the pretrained model which based on kinetics 400 to fine tune kinetics 600.
Because the classes number is different,so I ignore fc parameters during loading the pretrained caffe2 model.When I try to train the network,several problems occur.
1 distributed train,nan appear in s2 or s3 .I thought maybe data wrong ,but is not.When i use single
gpu,all things is ok
2 I have trained for 2-3 weeks,loss decreases slowly and top1 error is about 45-47% ,top 5 error is about 25-27%,can you provide any suggestion?
Thanks a lot!

**Comment by takatosp1** [comment]
Hi,
Thanks for using the codebase!
Short Answer (to anyone that considering to play with K600): 
I can release scripts and pretrain models for K600, but it might not be our top priority. If you want to play with K600, I suggest training it from scratch following with the setting we referred in our paper.

Long answer (maybe only useful to @chyueG)
Basically you want to play with the codebase on a new task (dataset): which is finetune on K600 pretrain model with K400 pretraining.

(I am not entirely sure what is your meta task, and not sure what is the reason/ goal of finetune with K400 pretraining on K600) The first thing I want to mention is that if you want to report the performance of your model, I think you should probably training from scratch on K600. 

When you decided to finetune on K600 with K400 pretraining, there are several things you need to do:
1) Make sure the there is no overlapping between K400 training set and K600 val/ testing set.
2) Find a new recipe for finetine. It is very likely the default learning rate is not applicable (normally you want to try with a smaller lr), and probably the optimal learning rate schedule is different accordingly. I am not sure how do you familiar with it, so I am not sure how much should I explain. Let me know if you think you might need more info regarding to what I described above.

Regarding to your question, again, I suggest you to train model on K600 from scratch rather than finetune with K400 pretraining.
For question 1), considering you are trying some of your own task on a new datasets, I need to know more details in order to help you on it. For example, how did you preprocess your K600 dataset, what is the mean and std you have for the dataset, what is the number of shards and number of GPUs did you use, what is the learning rate, when did you see the NaN (how many epochs). For a short answer, if you have it running fine with 1 GPU, then dist training is not likely to be the reason of NaN, I think you did something wrong somewhere else. (because the equivalent learning rate is smaller, I assume you haven't scale your learning rate)
For question 2), sorry I am not sure how did you train your model, and I don't know how long did you train with your model. Only knowing 2-3 weeks basically means little, I need to know what is the learning rate scheduling, what is the init learning rate, what is the model you train, and how many epoch you have trained it.
I am looking forward to having more info, then I can give you some suggestions.

**Comment by takatosp1** [comment]
I have opened the issue #50 for reference to users who is looking for K600 pretraining model/ recipe, so I will close this issue. 

Please feel free to share more details regarding to your task if you need any kind of suggestions regarding to you task.

**Comment by chyueG** [comment]
Thanks for your reply.I will do some experiments about several advice  you proposed  above.
